pkg/awsutils/awsutils.go

```diff
@@ -343,21 +356,71 @@ func (cache *EC2InstanceMetadataCache) initWithEC2Metadata() error {
 	}
 	log.Debugf("Found vpc-ipv4-cidr-block: %s ", cache.vpcIPv4CIDR)
 
-	// retrieve vpc-ipv4-cidr-blocks
+	// spawing go routine and retrieving vpc-ipv4-cidr-block
+	waitSGCh  := make(chan string)
+	waitVPCCh := make(chan string)
+	errorCh := make(chan error)
+	go wait.Forever(func() {
+		cache.initSGIDs(mac, waitSGCh, errorCh)
+		cache.initVPCIPv4CIDRs(mac, waitVPCCh, errorCh)
+	}, 30*time.Second)
+
+	//blocks until there's data available on waitSGCh and waitVPCCh or errorCh
+	for {
+		select {
+		case <-waitSGCh:
+			break
+		case errs := <-errorCh:
+			return errs
+		}
+		select {
+		case <- waitVPCCh:
+			break
+		case errs := <-errorCh:
+			return errs
+		}
+		break
```

Comment:
This infinite for-loop still doesn't look right. For example, the `break` on line 371 will exit the for-loop even if the one on line 378 has not yet exited. Options could be to have an [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup?tab=doc#example-Group-Parallel) or use something similar to [this  `WaitGroup` example](https://golangcode.com/errors-in-waitgroups/).

I think an easier solution would be to remove the channels completely, call the functions sequentially, once each without a go routine in the `init()` function, then set up the background go routines after. What do you think?

Reply:
> This infinite for-loop still doesn't look right. For example, the `break` on line 371 will exit the for-loop even if the one on line 378 has not yet exited. 

Hmm, AFAIT, it will exit the first `select` and moves on to the second one, but not the `for-loop` itself. 

>Options could be to have an [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup?tab=doc#example-Group-Parallel) or use something similar to [this `WaitGroup` example](https://golangcode.com/errors-in-waitgroups/).

I tried both`errgroup` and `waitgroup` packages but had a difficult with `wait.Forever`

> I think an easier solution would be to remove the channels completely, call the functions sequentially, once each without a go routine in the `init()` function, then set up the background go routines after. What do you think?

Generally, I would like to avoid `init()` function but like the idea to call to function and spawn separate `go-routines` for both of them and don't worry about channels. I am open to ideas, let me know :)